### PR TITLE
Fix Failing Casing Tests with Invariant Culture

### LINKED ISSUE #39522: `System.Globalization.Nls.Tests.ToUpper/ToLower_netcore` test failures in local environment
### Description

Run tests in `System.Globalization.Nls.Tests` test project. The characters after the ToLower/ToUpper call do not change even though we expect it to change.
```
Test	Duration	Traits	Error Message
System.Globalization.Tests.TextInfoMiscTests.ToLower_Netcore Failed	24 ms		Assert.Equal() Failure            ↓ (pos 1) Expected: 𐐨 Actual:   𐐀            ↑ (pos 1)
System.Globalization.Tests.TextInfoMiscTests.ToUpper_netcore Failed	< 1 ms		Assert.Equal() Failure            ↓ (pos 1) Expected: 𐐀 Actual:   𐐨            ↑ (pos 1)

```

Please note that this only seems to happen when no culture name is specified (`new CultureInfo("")`); the exact same test but with the culture name `en-US` / `fr` / `fr-FR` passes fine.

https://github.com/dotnet/runtime/blob/f4e9146d1d2ba7cf48ab59022658619fc9dbaec3/src/libraries/System.Globalization/tests/System/Globalization/TextInfoTests.cs#L174-L184

### Configuration

* Windows 10 Pro Insider Preview, Build 20170 (Also noticed in a couple previous flights), x64
* Locale ko-KR
* Ran with VS2019 16.7 P4, with `./build.cmd -vs System.Globalization -runtimeConfiguration Release -librariesConfiguration Debug`

### Other information

Possibly a Windows NLS issue, rather than the libraries implementation being wrong? Using builds from the dev channel probably doesn't help a lot 😅

### PR DESCRIPTION
**Pull request by tarekgh**
Fixes https://github.com/dotnet/runtime/issues/39522
The test was failing because of a regression on Windows side.

### CODE REVIEW

File: src/libraries/System.Runtime/tests/System/Text/RuneTests.cs [hunk 1]
```diff
@@ -50,8 +50,6 @@ public static void Casing_CultureAware(int original, int upper, int lower, strin
         [InlineData('\u0130', '\u0130', '\u0130')] // U+0130 LATIN CAPITAL LETTER I WITH DOT ABOVE
         [InlineData('\u0131', '\u0131', '\u0131')] // U+0131 LATIN SMALL LETTER DOTLESS I
         [InlineData('\u1E9E', '\u1E9E', '\u1E9E')] // U+1E9E LATIN CAPITAL LETTER SHARP S
-        [InlineData(0x10400, 0x10400, 0x10428)] // U+10400 DESERET CAPITAL LETTER LONG I
```
Comment (safern) [inline src/libraries/System.Runtime/tests/System/Text/RuneTests.cs:53]: Should we move this to `MemberData` and condition these two test cases to run when `!NlsGlobalization`

**Review by safern — APPROVED** [review]
>  because of a regression on Windows side.

Were you able to follow up with Windows?

File: src/libraries/System.Runtime/tests/System/Text/RuneTests.cs (hunk 1 above)
Comment (tarekgh) [reply src/libraries/System.Runtime/tests/System/Text/RuneTests.cs:53]: @GrabYourPitchforks can advise about that as I am not sure why he added this test cases to Nls only.

**Comment by tarekgh** [comment]
> Were you able to follow up with Windows?

Yes. I even spent some time debugging it in Windows side as they somehow couldn't easily repro it. Possibly because of some more customization of the version they are using. Windows team still going doing more investigation on that.

File: src/libraries/System.Runtime/tests/System/Text/RuneTests.cs (hunk 1 above)
Comment (GrabYourPitchforks) [reply src/libraries/System.Runtime/tests/System/Text/RuneTests.cs:53]: Probably laziness on my part. :)

**Review by GrabYourPitchforks — APPROVED** [review]

File: src/libraries/System.Runtime/tests/System/Text/RuneTests.cs (hunk 1 above)
Comment (tarekgh) [reply src/libraries/System.Runtime/tests/System/Text/RuneTests.cs:53]: will add it then :-)

**Comment by GrabYourPitchforks** [comment]
Reminder: we sometimes have ancient versions of ICU floating around, and it can cause unexpected test failures if we exercise scripts that were introduced within the past few years. See https://github.com/dotnet/runtime/pull/1514#issuecomment-617551026 for some context.

**Comment by tarekgh** [comment]
> Reminder: we sometimes have ancient versions of ICU floating around, and it can cause unexpected test failures if we exercise scripts that were introduced within the past few years. See #1514 (comment) for some context.

Yes, we still support ICU version down to v50. but let's see if we run into such issue with this test and then we can fix it.

**Comment by tarekgh** [comment]
The CI finished successfully but for some reason showing one of the tasks still running.

**Pull request merged**